pkg/virt-handler/vm.go

```diff
@@ -2746,7 +2746,10 @@ func (d *VirtualMachineController) vmUpdateHelperMigrationTarget(origVMI *v1.Vir
 		}
 	}
 
-	options := virtualMachineOptions(nil, 0, nil, d.capabilities, disksInfo, d.clusterConfig)
+	options, err := virtualMachineOptionsWithNetworkConfig(nil, 0, nil, d.capabilities, disksInfo, d.clusterConfig)
```

Comment:
I am not clear why we need this for migration.
When migration occurs, I would expect the domain config is passed from the source to the destination through libvirt, without the need to kubevirt to redefine it.

Reply:
Well, theoretically we can avoid passing the `domainAttachments` to `SyncMigrationTarget` and everything, apparently, will  keep working since indeed the domain is not updated.
However, the `SyncMigrationTarget` flow is invoking "full" phase2, same as on VM startup. So the domain is fully generated although it is not updated.
One [example](https://github.com/kubevirt/kubevirt/blob/ce0308520d5a59a941c59dc54be5f56a8d26ce40/pkg/virt-launcher/virtwrap/live-migration-target.go#L96) usage of the generated domain is when invoking the `onDefineDomain` of the sidecars.
The sidecars may reside on the information in the generated domain.

Reply:
Thanks for the clarification.
If we are to change it, we should probably drop this in general. Futuristic.